### PR TITLE
CI: remove python3.9 support

### PR DESCRIPTION
**Pull request by skoudoro**
This PR removes python 3.9 support.

python3.10 is the new minimal version. 

CI's and pyproject.toml has been updated.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,9 +23,10 @@ classifiers = [
     'License :: OSI Approved :: BSD License',
     'Programming Language :: Python',
     'Programming Language :: Python :: 3',
-    'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
     'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
+    'Programming Language :: Python :: 3.13',
```
[the remark this answers is not in the document]
Comment (skoudoro) [reply pyproject.toml:29]: You are right, I was anticipating https://github.com/dipy/dipy/pull/3400.

but I should actually update it on #3400. I will update that

[the remark this answers is not in the document]
Comment (skoudoro) [reply pyproject.toml]: fix in [d503695](https://github.com/dipy/dipy/pull/3421/commits/d503695fe7bc7ebf7b803e2d1793ee6ef2e589a5)

File: doc/devel/toolchain.rst [hunk 2]
```diff
@@ -78,7 +78,7 @@ Python release.
  1.7.0              >=3.8, <=3.11               >=1.19.5, <1.24.0
  1.8.0              >=3.8, <=3.12               >=1.19.5, <1.26.0
  1.9.0              >=3.9, <3.12                >=1.21.6, <1.27.0
- 1.10.0             >=3.9, <3.13                >=1.21.6, <2.2.0
+ 1.10.0             >=3.10, <3.14                >=1.21.6, <2.2.0
```
Comment (arokem) [inline doc/devel/toolchain.rst:81]: Sorry: here too?

```suggestion
 1.10.0             >=3.10, <3.13                >=1.21.6, <2.2.0
```

**Comment by arokem** [comment]
Great! I believe this is ready to merge as soon as the CI comes back clean.

**Comment by skoudoro** [comment]
all green, read to be merged!

**Pull request merged**